mathjax3-ts/output/chtml/Wrappers/mtable.ts

```diff
@@ -0,0 +1,387 @@
+            for (const cell of row.childNodes) {
+                //
+                //  Get the left and right-hand spacing
+                //
+                const lspace = spacing[i++];
+                const rspace = spacing[i];
+                //
+                //  Set the style for the spacing, if it is needed, and isn't the
+                //  default already set in the mtd styles
+                //
+                const style = (cell ? cell.chtml : row.chtml.childNodes[i] as HTMLElement).style;
+                if ((i > 1 || frame) && lspace !== '.5em') {
+                    style.paddingLeft = lspace;
+                }
+                if ((i < this.numCols || frame) && rspace !== '.5em') {
+                    style.paddingRight = rspace;
+                }
+            }
+        }
+    }
+
+    /*
+     * Add borders to the left of cells to make the column lines
+     */
+    protected handleColumnLines() {
+        if (this.node.attributes.get('columnlines') === 'none') return;
+        const lines = this.getColumnAttributes('columnlines');
+        if (!lines) return;
+        for (const row of this.childNodes) {
+            let i = 0;
+            for (const cell of (Array.from(row.chtml.childNodes) as HTMLElement[]).slice(1)) {
+                const line = lines[i++];
+                if (line === 'none') continue;
+                cell.style.borderLeft = '.07em ' + line;
+            }
+        }
+    }
+
+    /*
+     * Add vertical alignment to rows, and override in the cells, if needed
+     */
+    protected handleRowAlign() {
+        const rowAlign = this.getRowAttributes('rowalign') || [];
+        let i = 0;
+        for (const row of this.childNodes) {
+            const align = (row.node.attributes.get('rowalign') as string) || rowAlign[i++];
+            if (align !== 'baseline') {
+                row.chtml.style.verticalAlign = align;
+            }
+            for (const cell of row.childNodes) {
+                const calign = cell.node.attributes.get('rowalign') as string;
+                if (calign && calign !== align) {
+                    cell.chtml.style.verticalAlign = calign;
+                }
+            }
+        }
+    }
+
+    /*
+     * Set the inter-row spacing for all rows
+     *  (Use frame spacing on the outsides, if needed, and use half the row spacing on each
+     *   neighboring row, so that if row lines are needed, they fall in the middle
+     *   of the row space.)
+     *
+     * @param{boolean} frame  Whether to include frame spacing on the top and bottom or not
+     * @param{string} fspace  The frame spacing to use, if any
+     */
+    protected handleRowSpacing(frame: boolean, fspacing: string) {
+        //
+        //  Get the row spacing values, and add the frame spacing values at the left and right
+        //
+        const spacing = this.convertLengths(this.getRowAttributes('rowspacing'), 2);
+        if (!spacing) return;
+        spacing.unshift(fspacing);
```

Comment:
What happens if there is no frame spacing?

Reply:
I've added code to make sure it is defined.